### ISSUE TITLE
Does moto support multiple accounts?

### ISSUE THREAD

**Issue by rtoma** [spulec/moto#4319]
Hi,

My code under test assumes roles in different AWS accounts. In each account resources are created.

Does moto support asserting resources in multiple accounts? If so, I have a hard time getting it to work. Please advice.

My simplified code under test `app.py`:

```python
#!/usr/bin/env python
import boto3
from boto3_type_annotations.sts import Client as StsClient
from boto3_type_annotations.ec2 import Client as Ec2Client

ACCOUNT_IDS = ['111111111111', '222222222222']


def assume_in_account(account_id: str) -> boto3.Session:
    sts: StsClient = boto3.client('sts')
    role_arn = f'arn:aws:iam::{account_id}:role/app'
    credentials = sts.assume_role(RoleArn=role_arn, RoleSessionName='foo')[
        'Credentials']
    session = boto3.Session(aws_access_key_id=credentials['AccessKeyId'],
                            aws_secret_access_key=credentials['SecretAccessKey'],
                            aws_session_token=credentials['SessionToken'])
    return session


def provision_account(account_id: str):
    session = assume_in_account(account_id)
    ec2: Ec2Client = session.client('ec2')
    ec2.create_key_pair(KeyName=f'key-{account_id}')


def main():
    for account_id in ACCOUNT_IDS:
        provision_account(account_id)


if __name__ == '__main__':
    main()
```

And my test code:

```python
from moto import mock_sts, mock_ec2
import app
from unittest.mock import patch
from app import assume_in_account as assume_in_account_original


@mock_sts
@mock_ec2
@patch('app.assume_in_account')
def test_app(assume_in_account_mock):

    session_per_account_id = {}

    def assume_in_account_wrapped(account_id: str):
        session = assume_in_account_original(account_id)
        session_per_account_id[account_id] = session
        return session

    assume_in_account_mock.side_effect = lambda account_id: assume_in_account_wrapped(
        account_id)

    app.main()

    assert len(session_per_account_id) == len(app.ACCOUNT_IDS)

    for account_id, session in session_per_account_id.items():
        ec2 = session.client('ec2')
        key_pairs = ec2.describe_key_pairs()['KeyPairs']
        assert len(key_pairs) == 1
        assert key_pairs[0]['KeyName'] == f'key-{account_id}'
```

Expected: no asserts failing.

Actual: the `assert len(key_pairs) == 1` fails, because the list contains 2 keys (1 for each account).

Thanks so much!

**Comment by bblommers** [comment]
Hi @rtoma, Moto does not, unfortunately. The entire framework was build with the assumption that there would be only a single user.

Will mark it as an enhancement though, as it would be a useful feature.

Related issue: #1339

**Comment by rtoma** [comment]
Thanks @bblommers for the quick answer. Is there a workaround / hack to still allow this? Like mocks per account? Anything to get me going? If so, :heart: to hear. TIA

**Comment by bblommers** [comment]
Not that I'm aware of @rtoma. Manually patching the method (@mock.patch) is probably your best bet for now in terms of easy of implementation/maintenance

**Comment by bblommers** [comment]
@rtoma Support for multiple accounts is now implemented, and is available as of Moto >= 4.0.0.dev3 - see the documentation here: http://docs.getmoto.org/en/latest/docs/multi_account.html

The full release schedule for Moto 4 can be found here: https://github.com/spulec/moto/issues/5366

Please let us know if you have any feedback.
